cmd/router/main.go

```diff
@@ -0,0 +1,165 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"math/rand"
+
+	"github.com/kserve/kserve/pkg/apis/serving/v1alpha1"
+	flag "github.com/spf13/pflag"
+)
+
+func callService(serviceUrl string, input []byte, res chan<- []byte) error {
+	resp, err := http.Post(serviceUrl, "application/json", bytes.NewBuffer(input))
+	if err != nil {
+		log.Fatalf("An error has occured %v", err)
+		return err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Fatalf("error while reading the response %v", err)
+		return err
+	}
+	res <- body
+	return nil
+}
+
+func pickupRoute(routes []v1alpha1.InferenceRoute) *v1alpha1.InferenceRoute {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	//generate num [0,100)
+	point := r.Intn(99)
+	start, end := 0, 0
+
+	for _, route := range routes {
+		start += end
+		end += int(*route.Weight)
+		if point >= start && point < end {
+			return &route
+		}
+	}
+	return nil
+}
+
+func routeStep(nodeName string, currentStep v1alpha1.InferenceRouter, graph v1alpha1.InferenceGraphSpec, input []byte, res chan<- []byte) error {
+	log.Printf("current step %v", nodeName)
+	response := map[string]interface{}{}
+	//For splitter and ABNTest call virtual service
+	if currentStep.RouterType == v1alpha1.Splitter {
+		route := pickupRoute(currentStep.Routes)
+		if route == nil {
+			return fmt.Errorf("The pickuped servcie is nil.")
+		}
+		result := make(chan []byte)
+		go callService(route.ServiceUrl, input, result)
+		responseBytes := <-result
+		var res map[string]interface{}
+		json.Unmarshal(responseBytes, &res)
+		response = res
+	} else if currentStep.RouterType == v1alpha1.Switch {
```

Comment:
This has not been implemented yet, we plan to use [k8s.io/client-go/util/jsonpath](http://k8s.io/client-go/util/jsonpath) to evaluate the jsonpath and route the request based on condition

Reply:
we are going to use gjson https://github.com/tidwall/gjson which is used by argo for condition check.